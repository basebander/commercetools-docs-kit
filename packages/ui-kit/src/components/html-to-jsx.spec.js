import React from 'react';
import { render } from '@testing-library/react';
import HtmlToJsx from './html-to-jsx';

describe('converting an HTML string to JSX', () => {
  const codeExamples = [
    `<span class="red">Hello</span>`,
    `<div class="red">
      <span>Hello</span>
      <span class="bold">World</span>
    </div>`,
    `<div data-testid="foo">
      <span>Hello</span>
      <span> </span>
      <span>World</span>
    </div>`,
  ];
  codeExamples.forEach(code => {
    it(`should render React component for ${code}`, () => {
      const rendered = render(<HtmlToJsx value={code} />);
      expect(rendered.container.firstChild).toMatchSnapshot();
    });
  });
});
