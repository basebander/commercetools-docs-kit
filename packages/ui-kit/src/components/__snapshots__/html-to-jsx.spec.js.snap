// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converting an HTML string to JSX should render React component for <div class="red">
      <span>Hello</span>
      <span class="bold">World</span>
    </div> 1`] = `
<div
  class="red"
>
  
      
  <span>
    Hello
  </span>
  
      
  <span
    class="bold"
  >
    World
  </span>
  
    
</div>
`;

exports[`converting an HTML string to JSX should render React component for <div data-testid="foo">
      <span>Hello</span>
      <span> </span>
      <span>World</span>
    </div> 1`] = `
<div
  data-testid="foo"
>
  
      
  <span>
    Hello
  </span>
  
      
  <span>
     
  </span>
  
      
  <span>
    World
  </span>
  
    
</div>
`;

exports[`converting an HTML string to JSX should render React component for <span class="red">Hello</span> 1`] = `
<span
  class="red"
>
  Hello
</span>
`;
